streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignorTest.java

```diff
@@ -715,23 +1087,49 @@ private Map<UUID, Map<String, Optional<String>>> getRandomProcessRacks(final int
         return processRacks;
     }
 
-    private SortedMap<TaskId, Set<TopicPartition>> getTaskTopicPartitionMap(final int tpSize) {
+    private SortedMap<TaskId, Set<TopicPartition>> getTaskTopicPartitionMap(final int tpSize, final boolean changelog) {
         final SortedMap<TaskId, Set<TopicPartition>> taskTopicPartitionMap = new TreeMap<>();
+        final String topicName = changelog ? CHANGELOG_TOPIC_PREFIX : TOPIC_PREFIX;
         for (int i = 0; i < tpSize; i++) {
-            taskTopicPartitionMap.put(new TaskId(i, 0), mkSet(new TopicPartition("topic" + i, 0)));
+            taskTopicPartitionMap.put(new TaskId(i, 0), mkSet(
+                new TopicPartition(topicName + i, 0),
+                new TopicPartition(topicName + (i + 1) % tpSize, 0)
```

Comment:
Why do we add two partitions (from different topics) per task now? And why `(i + 1) % tpSize` ?

Reply:
Just to make it more complicated. There's some code to use flatmap, want to make sure it works as expected.